### ISSUE TITLE
'Disabling PEP 517 processing is invalid' error when creating dev environment

### ISSUE THREAD

**Issue by toporowska** [pandas-dev/pandas#53315]
I followed the documentation regarding creating a development environment: https://pandas.pydata.org/docs/development/contributing_environment.html

In step 3, while running
```
python -m pip install -e . --no-build-isolation --no-use-pep517
```
 I got this error:
```
--no-use-pep517 error: It is not possible to use --no-use-pep517
without setuptools and wheel installed.
```
I installed setuptools and wheel with pip and got the following:
```
WARNING: Ignoring invalid distribution -yclvm (c:\users\user\appdata\local\packages\pythonsoftwarefoundation.python.3.10_qbz5n2kfra8p0\localcache\local-packages\python310\site-packages)
Obtaining file:///C:/Users/user/Desktop/pycon%20sprint/pandas-barbara
ERROR: Disabling PEP 517 processing is invalid: project specifies a build backend of mesonpy in pyproject.toml
```
Other than installing wheel and setuptools I followed the instructions provided in the documentation.

Environment info:

- Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)] on win32
- pip 23.1.2
- Windows 10

**Comment by datapythonista** [comment]
@lithomas1 any idea on where can the problem be? I guess the docs are not updated and the `--no-use-pep517`, but that fails with an error that `mesonpy` is not installed, and when trying to install `mesonpy` it fails as it seems like mesonpy's `setup.py` is expecting a `requirements.txt` file, which I don't fully understand.

**Comment by MarcoGorelli** [comment]
hey - you should follow the instructions here https://pandas.pydata.org/docs/dev/development/contributing_environment.html

**Comment by datapythonista** [comment]
Good catch @MarcoGorelli, I didn't realize, thank you. Maybe we should eventually reopen the discussion on why the development documentation is under Sphinx and not in the website, where we wouldn't have the problem of people checking an obsolete version...

**Comment by lithomas1** [comment]
Yeah, you can't use ``--no-use-pep517`` if you specify a build backend(in our case, it's meson).

This is the second time that this has happened with people reading the old devdocs, so ideally I'd like to get this fixed.

@datapythonista Any ideas on potential fixes?
(Maybe we can avoid building the contributing/development sections on the release docs?)

**Comment by datapythonista** [comment]
I think the things that are not version specific should live in the website, that's why we have a website and not just Sphinx. I tried to move the dev docs, the roadmap and the ecosystem to the website when we built it, but there were objections that I couldn't understand. Surely happy to move them now, and if there is agreement, happy to organize a sprint to convert the files from rst to markdown (I think they are rst, otherwise the change is trivial).
